contracts/governance/ProposalValidator.sol

```diff
@@ -3,198 +3,253 @@ pragma solidity 0.7.6;
+    return strategy.validateProposalCreation(creator, startTime, endTime);
   }
 
   /**
-   * @dev Returns whether a user has enough Proposition Power to make a proposal.
-   * @param governance Governance Contract
-   * @param user Address of the user to be challenged.
-   * @param blockNumber Block Number against which to make the challenge.
-   * @return true if user has enough power
+   * @dev Called to validate a generic proposal
+   * @param strategy votingPowerStrategy contract to calculate voting power
+   * @param creator address of the creator
+   * @param startTime timestamp when vote starts
+   * @param endTime timestamp when vote ends
+   * @param options list of proposal vote options
+   * @param daoOperator address of daoOperator
+   * @return boolean, true if can be created
    **/
-  function isPropositionPowerEnough(
-    IKyberGovernance governance,
-    address user,
-    uint256 blockNumber
-  ) public override view returns (bool) {
-    IGovernanceStrategy currentGovernanceStrategy = IGovernanceStrategy(
-      governance.getGovernanceStrategy()
-    );
-    return
-      currentGovernanceStrategy.getPropositionPowerAt(user, blockNumber) >=
-      getMinimumPropositionPowerNeeded(governance, blockNumber);
+  function validateGenericProposalCreation(
+    IVotingPowerStrategy strategy,
+    address creator,
+    uint256 startTime,
+    uint256 endTime,
+    string[] calldata options,
+    address daoOperator
+  ) external view returns (bool) {
+    // check authorization
+    if (creator != daoOperator) return false;
+    // check vote duration
+    if (endTime.sub(startTime) < MIN_VOTING_DURATION) return false;
+    // check options length
+    if (options.length > MAX_VOTING_OPTIONS) return false;
+
+    return strategy.validateProposalCreation(creator, startTime, endTime);
   }
 
   /**
-   * @dev Returns the minimum Proposition Power needed to create a proposition.
-   * @param governance Governance Contract
-   * @param blockNumber Blocknumber at which to evaluate
-   * @return minimum Proposition Power needed
+   * @dev Called to validate the cancellation of a proposal
+   * @param governance governance contract to fetch proposals from
+   * @param proposalId Id of the generic proposal
+   * @param user entity initiating the cancellation
+   * @return boolean, true if can be cancelled
    **/
-  function getMinimumPropositionPowerNeeded(IKyberGovernance governance, uint256 blockNumber)
-    public
-    override
-    view
-    returns (uint256)
-  {
-    IGovernanceStrategy currentGovernanceStrategy = IGovernanceStrategy(
-      governance.getGovernanceStrategy()
+  function validateProposalCancellation(
+    IKyberGovernance governance,
+    uint256 proposalId,
+    address user
+  ) external view returns (bool) {
+    IKyberGovernance.ProposalWithoutVote memory proposal = governance.getProposalById(proposalId);
+    return (
+      strategy.getVotingPower(user, timestamp) >= 
```

Comment:
where are strategy, timestamp getting from? We don't have `PROPOSITION_THRESHOLD` for this simple strategy, so may can just return false here